service/controller/resource/resource.go

```diff
@@ -44,14 +45,35 @@ func New(config Config) ([]resource.Interface, error) {
 		}
 	}
 
-	var certificatesResource resource.Interface
+	var apiCertificatesResource resource.Interface
 	{
 		c := certificates.Config{
-			K8sClient: config.K8sClient,
-			Logger:    config.Logger,
+			Name:                "api-certificates",
+			K8sClient:           config.K8sClient,
+			Logger:              config.Logger,
+			SourceNameFunc:      key.Namespace,
```

Comment:
This sounds funny, but this function just returns `<cluster id>-prometheus` which is exactly what we want here. Maybe renaming it to `key.PrometheusSuffix` would make sense.

Reply:
I agree we should rename it but maybe not prometheus suffix as this name can be used for every cluster. Maybe PrometheusName or PrometheusInstanceName would make sense?